lms/djangoapps/discussion/rest_api/discussions_notifications.py

```diff
@@ -0,0 +1,237 @@
+        """
+        Get parent response object
+        """
+        if self.parent_id and not self.parent_response:
+            self.parent_response = Comment(id=self.parent_id).retrieve()
+
+        return self.parent_response
+
+    def send_new_response_notification(self):
+        """
+        Send notification to users who are subscribed to the main thread/post i.e.
+        there is a response to the main thread.
+        """
+        if not self.parent_id and self.creator.id != int(self.thread.user_id):
+            self._send_notification([self.thread.user_id], "new_response")
+
+    def _response_and_thread_has_same_creator(self) -> bool:
+        """
+        Check if response and main thread have same author.
+        """
+        return int(self.parent_response.user_id) == int(self.thread.user_id)
+
+    def send_new_comment_notification(self):
+        """
+        Send notification to parent thread creator i.e. comment on the response.
+        """
+        #
+        if (
+            self.parent_response and
+            self.creator.id != int(self.thread.user_id)
+        ):
+            # use your if author of response is same as author of post.
+            author_name = "your" if self._response_and_thread_has_same_creator() else self.parent_response.username
+            context = {
+                "author_name": author_name,
+            }
+            self._send_notification([self.thread.user_id], "new_comment", extra_context=context)
+
+    def send_new_comment_on_response_notification(self):
+        """
+        Send notification to parent response creator i.e. comment on the response.
+        Do not send notification if author of response is same as author of post.
+        """
+        if (
+            self.parent_response and
+            self.creator.id != int(self.parent_response.user_id) and not
+        self._response_and_thread_has_same_creator()
+        ):
+            self._send_notification([self.parent_response.user_id], "new_comment_on_response")
+
+    def send_response_on_followed_post_notification(self):
+        """
+        Send notification to followers of the thread/post
+        except:
+        Tread creator , response creator,
+        """
+        users = []
+        page = 1
+        has_more_subscribers = True
+
+        while has_more_subscribers:
+
+            subscribers = Subscription.fetch(self.thread.id, query_params={'page': page})
+            if page <= subscribers.num_pages:
+                for subscriber in subscribers.collection:
+                    # Check if the subscriber is not the thread creator or response creator
+                    subscriber_id = int(subscriber.get('subscriber_id'))
+                    # do not send notification to the user who created the response and the thread
+                    if subscriber_id != int(self.thread.user_id) and subscriber_id != int(self.creator.id):
+                        users.append(subscriber_id)
```

Comment:
Can we have a function for this check, instead of checking this condition for every notification type!
Something like:
```python
def is_author(thread, user):
...
```